src/adaXT/predict/predict.pyx

```diff
@@ -191,19 +189,28 @@ cdef class PredictRegression(Predict):
 
     @staticmethod
     def forest_predict(predictions: np.ndarray, **kwargs):
-        return np.mean(predictions, axis=1)
+        return np.mean(predictions, axis=-1)
+
+
+cdef class PredictLocalPolynomial(PredictRegression):
 
-cdef class PredictLinearRegression(PredictRegression):
     def predict(self, object X, **kwargs):
         cdef:
-            int i, cur_split_idx, n_obs
+            int i, cur_split_idx, n_obs, ind
             double cur_threshold
             object cur_node
-            double[:] Y
+            double[:, ::1] deriv_mat
+
+        if "order" not in kwargs.keys():
+            order = [0, 1, 2]
+        else:
+            order = np.array(kwargs['order'], ndmin=1, dtype='int')
```

Comment:
might be an idea to add a check that size < 3. Currently we allow for duplicates. Not necessary.

Reply:
done